### DISCUSSION TITLE
Big Sur 11.1 ruby-build failed 2.5.1

### DISCUSSION THREAD

**Discussion by emschult** [rbenv/ruby-build#1931] [Build failures]
I am trying to download ruby 2.5.1 and have repeatedly run into failures. I have tried a number of trouble shooting techniques. Any help would be greatly appreciated!!

_____@____-MacBook-Pro ~ % rbenv install 2.5.1
Downloading openssl-1.1.1i.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/e8be6a35fe41d10603c3cc635e93289ed00bf34b79671a3a4de64fcee00d5242
Installing openssl-1.1.1i...

BUILD FAILED (macOS 11.1 using ruby-build 20201225)

Inspect or clean up the working tree at /var/folders/8q/3_kkmpr9335_5lbl29t0tnfh0000gn/T/ruby-build.20210111172328.5336.p7cDAo
Results logged to /var/folders/8q/3_kkmpr9335_5lbl29t0tnfh0000gn/T/ruby-build.20210111172328.5336.log

Last 10 log lines:
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL file first)         ***
***                                                                ***
**********************************************************************
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun

**Comment by hsbt** [comment]
Try with https://github.com/rbenv/ruby-build/issues/1505#issuecomment-752274233

**Comment by emschult** [comment]
@hsbt 
Thank you for commenting. I followed your guidance and ran:

$ sudo rm -rf /Library/Developer/CommandLineTools
$ xcode-select --install
$ sudo xcode-select -s /Library/Developer/CommandLineTools

It ran without issues. I tried to download again using

rbenv install 2.5.1 and got a failure again:

erikschulte@Eriks-MacBook-Pro ~ % sudo rm -rf /Library/Developer/CommandLineTools
Password:
erikschulte@Eriks-MacBook-Pro ~ % xcode-select --install
xcode-select: note: install requested for command line developer tools
erikschulte@Eriks-MacBook-Pro ~ % rbenv install 2.5.1
Downloading openssl-1.1.1i.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/e8be6a35fe41d10603c3cc635e93289ed00bf34b79671a3a4de64fcee00d5242
Installing openssl-1.1.1i...
Installed openssl-1.1.1i to /Users/erikschulte/.rbenv/versions/2.5.1

Downloading ruby-2.5.1.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.5/ruby-2.5.1.tar.bz2
Installing ruby-2.5.1...

WARNING: ruby-2.5.1 is nearing its end of life.
It only receives critical security updates, no bug fixes.

ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.1 using ruby-build 20201225)

Inspect or clean up the working tree at /var/folders/8q/3_kkmpr9335_5lbl29t0tnfh0000gn/T/ruby-build.20210111212940.86916.MlqKFS
Results logged to /var/folders/8q/3_kkmpr9335_5lbl29t0tnfh0000gn/T/ruby-build.20210111212940.86916.log

Last 10 log lines:
3 warnings generated.
compiling ../.././ext/psych/yaml/reader.c
compiling ../.././ext/psych/yaml/emitter.c
compiling ../.././ext/psych/yaml/parser.c
linking shared-object date_core.bundle
5 warnings generated.
linking shared-object zlib.bundle
1 warning generated.
linking shared-object psych.bundle
make: *** [build-ext] Error 2
erikschulte@Eriks-MacBook-Pro ~ % vim /var/folders/8q/3_kkmpr9335_5lbl29t0tnfh0000gn/T/ruby-build.20210111212940.86916.log


Any ideas?? Thank you for being willing to help!

**Comment by hsbt** [comment]
> Any ideas?? Thank you for being willing to help!

Ruby 2.5 is not supported on M1 macOS.

**Reply by aakashkaji** [reply]
> > Any ideas?? Thank you for being willing to help!
> 
> Ruby 2.5 is not supported on M1 macOS.

Actually you have to install natively
You try this command to install 2.5.x versions
-$  RUBY_CFLAGS=-DUSE_FFI_CLOSURE_ALLOC arch -arm64 rbenv install 2.5.3

**Comment by emschult** [comment]
@hsbt 
Do you know of any Ruby builds that are compatible?

I was planning to use this computer to do App Academy. Any idea if there is a good syntax change cheat sheet between versions?

Thank you for all the help!!

**Comment by hsbt** [comment]
> Do you know of any Ruby builds that are compatible?

Ruby 2.7.2+ and Ruby 3.0.0.

**Comment by emschult** [comment]
@hsbt 
Thank you very much for the help! After a bit more troubleshooting I was able to successfully install:
2.7.2 & 3.0.0

Excited to start learning. Thank you again!!

**Comment by misraelson** [comment]
For any beings from the future, use this command to install on M1 Mac (I'm using asdf but works for rbenv as well):
`RUBY_CFLAGS="-Wno-error=implicit-function-declaration" asdf install ruby 2.5.1`
via: Prabin Poudel
https://dev.to/truemark/fix-issue-while-installing-ruby-with-rbenv-in-m1-mac-3dfd
